tests/bgp/test_bgp_command.py

```diff
@@ -0,0 +1,88 @@
+import pytest
+import logging
+import re
+
+from tests.common.helpers.assertions import pytest_assert
+
+
+pytestmark = [
+    pytest.mark.topology("t0", "t1", "m0", "mx"),
+    pytest.mark.device_type('vs')
+]
+
+logger = logging.getLogger(__name__)
+
+
+# Function to parse the "Displayed X routes and Y total paths" line
+def parse_routes_and_paths(output):
+    match = re.search(r"Displayed\s+(\d+)\s+routes\s+and\s+(\d+)\s+total paths", output)
+    if match:
+        routes = int(match.group(1))
+        paths = int(match.group(2))
+        return routes, paths
+    return None
+
+
+@pytest.mark.parametrize("ip_version", ["ipv4", "ipv6"])
+def test_bgp_network_command(
+    duthosts, enum_rand_one_per_hwsku_frontend_hostname, ip_version
+):
+    """
+    @summary: This test case is to verify the output of "show ip bgp network" command
+    if it matches the output of "docker exec -i bgp vtysh -c show bgp ipv4 all" command
+    """
+    duthost = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
+    if ip_version == "ipv4":
+        bgp_network_cmd = "show ip bgp network"
+        bgp_docker_cmd = 'docker exec -i bgp vtysh -c "show bgp ipv4 all"'
+    elif ip_version == "ipv6":
+        bgp_network_cmd = "show ipv6 bgp network"
+        bgp_docker_cmd = 'docker exec -i bgp vtysh -c "show bgp ipv6 all"'
+
+    bgp_network_output = duthost.shell(bgp_network_cmd)["stdout"]
+    pytest_assert(
+        "*=" in bgp_network_output, "Failed to run '{}' command".format(bgp_network_cmd)
+    )
+
+    bgp_docker_output = duthost.shell(bgp_docker_cmd)["stdout"]
+    pytest_assert(
+        "*=" in bgp_docker_output,
+        "Failed to run '{}' command".format(bgp_docker_output),
+    )
+    # Remove the first two lines from the docker command output
+    bgp_docker_output_lines = bgp_docker_output.splitlines()[2:]
+    bgp_docker_output_modified = "\n".join(bgp_docker_output_lines)
+
+    pytest_assert(
```

Comment:
It is possible that there could be cases where bgp docker is down on the DUT. Do we need to consider that as a test failure for this scenario? Meaning, if our intention is to check the correctness b/w the two commands, we can fail the test only if docker output is successful and network output failed.  If both failed, then it's not related to this test. What do you think?

Reply:
@prsunny If bgp docker is down, there will be no "*=" in the output, and case will fail. I can also add a check to see if rc equals 0. Yes, we will consider it as case failure, sanity check passes but here bgp docker container is down, that could be some problem and case failure can cause our attention.
My intention is to capture not only the output of show ip bgp network, but also check the bgp container health.

Reply:
ok